src/List.mo

```diff
@@ -459,29 +552,30 @@ module {
   public func removeLast<T>(list : List<T>) : ?T {
     var elementIndex = list.elementIndex;
     if (elementIndex == 0) {
-      shrink_index_block_if_needed(list);
-
       var blockIndex = list.blockIndex;
       if (blockIndex == 1) {
         return null
       };
+
+      shrinkIndexBlockIfNeeded(list);
```

Comment:
Why do we call this here and not after the line 
blockIndex -= 1;
?

Reply:
I see. You intentionally do shrinkIndexlockIfNeeded on the old indices so that one empty data block can remain at the end. Ok